packages/shared_preferences/shared_preferences_ios/ios/Classes/messages.g.m

```diff
@@ -0,0 +1,228 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+// Autogenerated from Pigeon (v1.0.16), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+#import "messages.g.h"
+#import <Flutter/Flutter.h>
+
+#if !__has_feature(objc_arc)
+#error File requires ARC to be enabled.
+#endif
+
+static NSDictionary<NSString *, id> *wrapResult(id result, FlutterError *error) {
+  NSDictionary *errorDict = (NSDictionary *)[NSNull null];
+  if (error) {
+    errorDict = @{
+      @"code" : (error.code ? error.code : [NSNull null]),
+      @"message" : (error.message ? error.message : [NSNull null]),
+      @"details" : (error.details ? error.details : [NSNull null]),
+    };
+  }
+  return @{
+    @"result" : (result ? result : [NSNull null]),
+    @"error" : errorDict,
+  };
+}
+
+@interface SharedPreferencesApiCodecReader : FlutterStandardReader
+@end
+@implementation SharedPreferencesApiCodecReader
+@end
+
+@interface SharedPreferencesApiCodecWriter : FlutterStandardWriter
+@end
+@implementation SharedPreferencesApiCodecWriter
+@end
+
+@interface SharedPreferencesApiCodecReaderWriter : FlutterStandardReaderWriter
+@end
+@implementation SharedPreferencesApiCodecReaderWriter
+- (FlutterStandardWriter *)writerWithData:(NSMutableData *)data {
+  return [[SharedPreferencesApiCodecWriter alloc] initWithData:data];
+}
+- (FlutterStandardReader *)readerWithData:(NSData *)data {
+  return [[SharedPreferencesApiCodecReader alloc] initWithData:data];
+}
+@end
+
+NSObject<FlutterMessageCodec> *SharedPreferencesApiGetCodec() {
+  static dispatch_once_t s_pred = 0;
+  static FlutterStandardMessageCodec *s_sharedObject = nil;
+  dispatch_once(&s_pred, ^{
+    SharedPreferencesApiCodecReaderWriter *readerWriter =
+        [[SharedPreferencesApiCodecReaderWriter alloc] init];
+    s_sharedObject = [FlutterStandardMessageCodec codecWithReaderWriter:readerWriter];
+  });
+  return s_sharedObject;
+}
+
+void SharedPreferencesApiSetup(id<FlutterBinaryMessenger> binaryMessenger,
+                               NSObject<SharedPreferencesApi> *api) {
+  {
+    FlutterBasicMessageChannel *channel = [FlutterBasicMessageChannel
+        messageChannelWithName:@"dev.flutter.pigeon.SharedPreferencesApi.remove"
+               binaryMessenger:binaryMessenger
+                         codec:SharedPreferencesApiGetCodec()];
+    if (api) {
+      NSCAssert([api respondsToSelector:@selector(removeKey:error:)],
+                @"SharedPreferencesApi api (%@) doesn't respond to @selector(removeKey:error:)",
+                api);
+      [channel setMessageHandler:^(id _Nullable message, FlutterReply callback) {
+        NSArray *args = message;
+        NSString *arg_key = args[0];
+        FlutterError *error;
+        NSNumber *output = [api removeKey:arg_key error:&error];
+        callback(wrapResult(output, error));
+      }];
+    } else {
+      [channel setMessageHandler:nil];
+    }
+  }
```

Comment:
Yea, that could be something we look into. It isn't boilerplate since it is generated and can be treated as a black box by the user.  Boilerplate to me means code a user has to manage that has semantic meaning that is small compared to its verbosity.